tracing-appender/src/rolling.rs

```diff
@@ -602,7 +602,28 @@ impl Inner {
                     return None;
                 }
 
-                let created = metadata.created().ok()?;
+                let created = metadata.created().ok().or_else(|| {
+                    let datetime = if let (Some(prefix), Some(suffix)) =
+                        (&self.log_filename_prefix, &self.log_filename_suffix)
+                    {
+                        let date = filename
+                            .split(prefix)
+                            .last()?
+                            .split(suffix)
+                            .next()?
+                            .strip_prefix('.')?
+                            .strip_suffix('.')?;
+
+                        PrimitiveDateTime::parse(date, &self.date_format)
+                            .ok()?
+                            .assume_utc()
+                    } else {
+                        PrimitiveDateTime::parse(filename, &self.date_format)
```

Comment:
What if the prefix is configured but the suffix is not?

Reply:
Oops, forgot that branch.